UI/system-info-posix.cpp

```diff
@@ -1,6 +1,623 @@
+		drm_path += "/subsystem_vendor";
+		file_str = os_quick_read_utf8_file(drm_path.c_str());
+		if (!file_str) {
+			dci->subsystem_vendor = 0;
+		} else {
+			// Skip over the "0x" and convert
+			sscanf(file_str + 2, "%x", &val);
+			dci->subsystem_vendor = val;
+			bfree(file_str);
+		}
+
+		// Get the subsystem_device
+		drm_path.resize(base_length);
+		drm_path += "/subsystem_device";
+		file_str = os_quick_read_utf8_file(drm_path.c_str());
+		if (!file_str) {
+			dci->subsystem_device = 0;
+		} else {
+			// Skip over the "0x" and convert
+			sscanf(file_str + 2, "%x", &val);
+			dci->subsystem_device = val;
+			bfree(file_str);
+		}
+
+		// Get the boot_vga flag
+		dci->boot_vga = false;
+		drm_path.resize(base_length);
+		drm_path += "/boot_vga";
+		file_str = os_quick_read_utf8_file(drm_path.c_str());
+		// 'boot_vga' is not present for some adapters (iGPU)
+		if (file_str) {
+			sscanf(file_str, "%d", &val);
+			if (val == 1)
+				dci->boot_vga = true;
+			bfree(file_str);
+		}
+
+		/* The amdgpu driver exports the GPU memory information
+		 * via sysfs nodes. Sadly NVIDIA doesn't have the same
+		 * information via sysfs. Read the amdgpu-based nodes
+		 * if present and get the required fields.
+		 */
+
+		// Get the GPU total dedicated VRAM, if available
+		drm_path.resize(base_length);
+		drm_path += "/mem_info_vram_total";
+		file_str = os_quick_read_utf8_file(drm_path.c_str());
+		if (!file_str) {
+			dci->dedicated_vram_total = 0;
+		} else {
+			sscanf(file_str, "%lu", &dci->dedicated_vram_total);
+			bfree(file_str);
+		}
+
+		// Get the GPU total shared system memory, if available
+		drm_path.resize(base_length);
+		drm_path += "/mem_info_gtt_total";
+		file_str = os_quick_read_utf8_file(drm_path.c_str());
+		if (!file_str) {
+			dci->shared_system_memory_total = 0;
+		} else {
+			sscanf(file_str, "%lu", &dci->shared_system_memory_total);
+			bfree(file_str);
+		}
+
+		blog(LOG_DEBUG,
+		     "%s: drm_adapter_info: PCI B:D:S:F: %s, device_id:0x%x,"
+		     "vendor_id:0x%x, sub_device:0x%x, sub_vendor:0x%x,"
+		     "boot_flag:%d, vram_total: %lu, sys_memory: %lu",
+		     __FUNCTION__, dci->dbsf.c_str(), dci->device_id, dci->vendor_id, dci->subsystem_device,
+		     dci->subsystem_vendor, dci->boot_vga, dci->dedicated_vram_total, dci->shared_system_memory_total);
+	} else {
+		return false;
+	}
+
+	return true;
+}
+
+static void adjust_gpu_model(std::string *model)
+{
+	/* Use the sub-string between the [] brackets. For example,
+	 * the NVIDIA Quadro P4000 model string from PCI ID database
+	 * is "GP104GL [Quadro P4000]", and we only want the "Quadro
+	 * P4000" sub-string.
+	 */
+	size_t first = 0;
+	size_t last = 0;
+	first = model->find('[');
+	last = model->find_last_of(']');
+	if ((last - first) > 1) {
+		std::string adjusted_model = model->substr(first + 1, last - first - 1);
+		model->assign(adjusted_model);
+	}
+}
+
+bool compare_match_strength(const drm_card_info &a, const drm_card_info &b)
```

Comment:
Unless this function is used in a different translation unit, might want to mark it as static.

Reply:
Also recommend checking other newly added non-exported functions elsewhere where this situation might apply, but this is just the one I noticed at first glance.

Reply:
> Unless this function is used in a different translation unit, might want to mark it as static.

Or for better C++-isms, put them into an anonymous (unnamed) namespace.

Reply:
I've put almost all functions in an anonymous namespace and dropped "static" for these. It's cleaner from perspective. Thanks for the suggestion, it makes sense.